packages/alea-frontend/pages/acl/[aclId].tsx

```diff
@@ -27,23 +35,56 @@ import MainLayout from '../../layouts/MainLayout';
 const AclId: NextPage = () => {
   const router = useRouter();
   const { query } = router;
-  const aclId = router.query.aclId as string;
+  const aclId = query.aclId as string;
 
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [deleteError, setDeleteError] = useState('');
   const [acls, setAcls] = useState<string[]>([]);
-  const [directMembersNamesAndIds, setDirectMembersNamesAndIds] = useState([]);
-  const [desc, setDesc] = useState<string>(null);
+  const [directMembersNamesAndIds, setDirectMembersNamesAndIds] = useState<any[]>([]);
+  const [desc, setDesc] = useState<string | null>(null);
   const [isOpen, setIsOpen] = useState<boolean>(false);
-  const [updaterACLId, setUpdaterACLId] = useState<string>(null);
+  const [updaterACLId, setUpdaterACLId] = useState<string | null>(null);
   const [userIsMember, setUserIsMember] = useState<boolean>(false);
   const [userIdInput, setUserIdInput] = useState<string>('');
   const [membershipStatus, setMembershipStatus] = useState<string>('');
   const [isUpdaterMember, setIsUpdaterMember] = useState<boolean>(false);
-  const [allMemberNamesAndIds, setAllMemberNamesAndIds] = useState([]);
+  const [allMemberNamesAndIds, setAllMemberNamesAndIds] = useState<any[]>([]);
   const [showAllMembers, setShowAllMembers] = useState<boolean>(false);
 
+  const [hasAssociatedResources, setHasAssociatedResources] = useState<boolean>(false);
+
+  const handleOpenDeleteDialog = () => {
+    setDeleteError('');
+
+    if (!hasAssociatedResources) {
+      setDeleteDialogOpen(true);
+    }
+  };
+
+  const handleDeleteConfirm = async () => {
+    try {
+      if (aclId) {
+        await deleteAcl(aclId);
+        setDeleteDialogOpen(false);
+        router.push('/acl');
+      }
+    } catch (e: any) {
+      console.error('Error deleting ACL:', e);
+      setDeleteError(
+        e.response?.data?.message ||
+          'Failed to delete ACL. It might be assigned to resources or have other dependencies.'
+      );
+    }
+  };
+
+  const handleDeleteCancel = () => {
+    setDeleteDialogOpen(false);
+    setDeleteError('');
+  };
+
   async function getMembers() {
     try {
-      const acl = await getAcl(aclId as string);
+      const acl = await getAcl(aclId);
       setDesc(acl?.description);
```

Comment:
```suggestion
     setDesc(acl?.description ?? null);
``` as the type defined is string or null